src/main/java/bob/Storage.java

```diff
@@ -0,0 +1,113 @@
+                newStorage.file.createNewFile();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return newStorage;
+    }
+
+    /**
+     * Loads all saved entries in the file associated with this Storage instance into a list.
+     *
+     * @return A TaskList loaded from the saved file.
+     * @throws FileNotFoundException File to be loaded from not found.
+     */
+    public TaskList load() {
+        List<Task> list = new ArrayList<>();
+
+        try {
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNextLine()) {
+                String nextLine = sc.nextLine();
+                String[] tokens = nextLine.strip().split("\\|");
+                String taskType = tokens[0];
+                boolean isDone = tokens[1].equals("1");
+                String taskDescription = tokens[2];
+
+                Task newTask;
+                switch (taskType) {
+                case ("T"):
+                    newTask = new ToDo(taskDescription, isDone);
+                    break;
+                case ("E"):
+                    String[] taskDescriptionTime = taskDescription.split("/at ");
+                    String eventTaskName = taskDescriptionTime[0];
+                    String time = taskDescriptionTime[1];
+                    newTask = new Event(eventTaskName, time, isDone);
+                    break;
+                case ("D"):
+                    String[] taskDescriptionSplit = taskDescription.split("/by ");
+                    String deadlineTaskName = taskDescriptionSplit[0];
+                    String deadline = taskDescriptionSplit[1];
+                    newTask = new Deadline(deadlineTaskName, deadline, isDone);
+                    break;
+                default:
+                    throw new CorruptedEntryException("Saved file has corrupted entries! :(");
+                }
+                list.add(newTask);
+            }
+        } catch (CorruptedEntryException e) {
+            System.out.println(e.getMessage());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return new TaskList(list);
+    }
+
+    /**
+     * Converts a given list of tasks into a String and writes it to the file associated with this Storage instance.
+     *
+     * @param tasks List of tasks to be written to file.
+     */
+    public void writeTaskListToFile(TaskList tasks) {
+        StringBuilder tasksString = new StringBuilder();
+        for (int i = 0; i < tasks.size(); i++) {
+            tasksString.append(tasks.get(i).generateSavedEntry());
+            tasksString.append("\n");
+        }
+        try {
+            FileWriter fileWriter = new FileWriter(file);
+            fileWriter.write(tasksString.toString());
+            fileWriter.close();
+        } catch (IOException e) {
+            System.out.println("Error writing to file!:( : " + e.getMessage());
+        }
+    }
+}
```

Comment:
I like that you have well-written javadocs, it makes code easier to read and understand!